### ISSUE TITLE
Feature request: Messaging DM contacts asking them to delete DM threads

### ISSUE THREAD

**Issue by amishrabbit** [timhutton/twitter-archive-parser#148]
Hi Tim. Great work on this project so far. 

I'm in a situation where I want to delete all my DM conversations, but as you probably know, the full deletion of DMs only happens when all parties in a DM message thread delete their messages.

There's no way to force others to delete their DMs, but I'd like to suggest a feature where the script can crawl the contents of the DM archive and then send a DM to each account that had previously DMed with my account. The message could say something like the following:

"I have downloaded my Twitter archive and I'm planning to delete all my direct messages, including this conversation, in __ days. I encourage you to do the same, since these messages will persist on Twitter's servers until and unless all parties to the DM thread delete their messages. I will no longer be responding to further messages. If you wish to reach me in the future you can contact me at ____ on [service]."

If I were adept at working in the Twitter API I would make this script myself, but I just don't have the skill to do so. I hope you'll consider adding this to a future update. Thanks.

**Comment by timhutton** [comment]
Hi @amishrabbit. Thanks. It's not just me though - many people have contributed to this project.

I wasn't aware that both parties needed to delete the DMs, thanks.

I can see a few problems with implementing this:
- To send an automated DM would require authenticating as you, which I don't think is straightforward.
- Even if both parties delete the DMs there is no guarantee that Twitter will delete them from the servers. They are not exactly behaving in a trustworthy fashion at the moment. If you have a legal requirement to delete the data then you'll need to contact Twitter directly.
- We'd be responsible for potentially spamming a lot of users.

**Comment by amishrabbit** [comment]
> * To send an automated DM would require authenticating as you, which I don't think is straightforward.

Yep, I can see that as a problem.  I think it can be done by customizing the script to use an API key, but I don't have the technical knowledge to know exactly how to do that.

> * Even if both parties delete the DMs there is no guarantee that Twitter will delete them from the servers. They are not exactly behaving in a trustworthy fashion at the moment. If you have a legal requirement to delete the data then you'll need to contact Twitter directly.

My understanding* is that these mechanisms are robust and take place within 30 days of the request**

* from being in personal contact with some of the newly-former sec ops tweeps
** as long as this code isn't being modified internally subsequent to the Great Layoffs

> * We'd be responsible for potentially spamming a lot of users.

I don't think that's avoidable. In fact, that's exactly what I'm suggesting, though I'd hesitate to call a "mass reply to existing conversations" spam in the strict sense.

**Comment by Bebetternow22** [comment]
Hi, I know Twitter keeps the messages on their servers even after we delete them.  How can I pull these if I have deleted the DM's on my end?  I am not sure if the other user has deleted theirs though.  Is there anyway to pull deleted direct messages from Twitter?  If so, could you help with this or send me some information on someone who can help?  I am new to coding, so I am kind of needing to rely on the expertise of others.  Thanks.
